### PR TITLE
feat(node): add http.request

### PR DESCRIPTION
**Pull request by kt3k**
part of #1703 
depends on #1706

### CODE REVIEW

File: node/internal/timers.ts [hunk 1]
```diff
@@ -75,3 +75,10 @@ export function getTimerDuration(msecs: number, name: string) {
 
   return msecs;
 }
+
+export default {
+  kTimeout,
+  TIMEOUT_MAX,
+  setUnrefTimeout,
+  getTimerDuration,
+}
```
Comment (bnoordhuis) [inline node/internal/timers.ts:84]:
```suggestion
};
```

File: node/internal/dtrace.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+} = (config.hasDtrace ? dtrace : {});
+
+export {
+  DTRACE_HTTP_CLIENT_REQUEST,
+  DTRACE_HTTP_CLIENT_RESPONSE,
+  DTRACE_HTTP_SERVER_REQUEST,
+  DTRACE_HTTP_SERVER_RESPONSE,
+  DTRACE_NET_SERVER_CONNECTION,
+  DTRACE_NET_STREAM_END,
+};
```
Comment (bnoordhuis) [inline node/internal/dtrace.ts:23]: I assume you copied this from node but we're never going to have dtrace support so you might as well simplify this file to always export stubs.

File: node/internal_binding/dtrace.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}
+export function DTRACE_HTTP_SERVER_RESPONSE() {
+  notImplemented();
+}
+export function DTRACE_NET_SERVER_CONNECTION() {
+  notImplemented();
+}
+export function DTRACE_NET_STREAM_END() {
+  notImplemented();
+}
```
Comment (bnoordhuis) [inline node/internal_binding/dtrace.ts:23]: And remove this file (maybe config.ts too)

**Review by bnoordhuis — APPROVED** [review]
Nice work!

**Comment by kt3k** [comment]
I'll continue working on a few more points:

- the server responds with an empty response with the code `res.write("foo"); res.end()`, which prevents some test cases passing.
- the client sometimes hangs with `socket hang up` error. e.g. in `test-http-client-get-url.js`, I still don't find the reason of this

**Review by bartlomieju — APPROVED** [review]
LGTM too, great work @kt3k

**Comment by kt3k** [comment]
Thanks for your reviews!

**Pull request merged**